pkg/logql/sketch/cms.go

```diff
@@ -6,19 +6,25 @@ import (
 )
 
 type CountMinSketch struct {
-	depth, width uint32
-	counters     [][]uint32
+	Depth, Width uint32
+	Counters     [][]uint32
 }
 
 // NewCountMinSketch creates a new CMS for a given width and depth.
 func NewCountMinSketch(w, d uint32) (*CountMinSketch, error) {
 	return &CountMinSketch{
-		depth:    d,
-		width:    w,
-		counters: make2dslice(w, d),
+		Depth:    d,
+		Width:    w,
+		Counters: make2dslice(w, d),
 	}, nil
 }
 
+func NewCountMinSketchFromErroAndProbability(epsilon float64, delta float64) (*CountMinSketch, error) {
+	width := math.Ceil(math.E / epsilon)
+	depth := math.Ceil(math.Log(1.0 / delta))
```

Comment:
See https://cs.stackexchange.com/questions/166937/what-is-the-proper-way-to-calculate-dimensions-of-count-min-sketch

Reply:
this is the calculation I used when doing all of our experiments, but I can't remember if I got good results using it

Reply:
I figured we have to stay here because this makes the matrix roughly the size of our max 10k vector.